buildSrc/src/main/groovy/org/elasticsearch/gradle/BuildPlugin.groovy

```diff
@@ -79,6 +78,30 @@ class BuildPlugin implements Plugin<Project> {
         configureConfigurations(project)
         project.ext.versions = VersionProperties.versions
         configureCompile(project)
+
+        String artifactsHost = project.ext.versions.elasticsearch.endsWith("-SNAPSHOT") ? "https://snapshots.elastic.co" : "https://artifacts.elastic.co"
+
+        project.afterEvaluate {
+            /*
+             * Order matters, the linksOffline for org.elasticsearch:elasticsearch must be the last one
+             * or all the links for the other packages (e.g org.elasticsearch.client) will point to core rather than their own artifacts
+             */
+            Closure sortClosure = { a, b -> b.group <=> a.group };
+            Closure depJavadocClosure = { dep ->
+                if (dep.group != null && dep.group.startsWith('org.elasticsearch')) {
+                    String substitution = project.ext.projectSubstitutions.get("${dep.group}:${dep.name}:${dep.version}")
+                    if (substitution != null) {
+                        project.javadoc.dependsOn substitution + ':javadoc'
+                        String artifactPath = dep.group.replaceAll('\\.', '/') + '/' + dep.name.replaceAll('\\.', '/') + '/' + dep.version
+                        String projectPath = substitution.replaceAll(':', '/')
```

Comment:
This isn't always true. You should use `project(substitution).path`.  But I don't think you need this at all. Instead of `"${project.rootDir}${projectPath}/build/docs/javadoc/"` below, use `"${project(substitution).buildDir}/docs/javadoc/"`